### PR TITLE
Boxpanel docs update

### PR DESCRIPTION
**Pull request by gochoam**
@sccolbert Documentation update.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,9 +4,9 @@ phosphor-boxpanel
 [![Build Status](https://travis-ci.org/phosphorjs/phosphor-boxpanel.svg)](https://travis-ci.org/phosphorjs/phosphor-boxpanel?branch=master)
 [![Coverage Status](https://coveralls.io/repos/phosphorjs/phosphor-boxpanel/badge.svg?branch=master&service=github)](https://coveralls.io/github/phosphorjs/phosphor-boxpanel?branch=master)
 
-A Phosphor layout panel which arranges its children into a single row or column.
-
-[API Docs](http://phosphorjs.github.io/phosphor-boxpanel/api/)
+This module provides a phosphor layout panel which arranges its children into a
+single row or column, making it possible to change the layout direction,
+spacing and length.
```
Comment (sccolbert) [inline README.md]: what do you mean by "length" here?

File: README.md [hunk 2]
```diff
@@ -117,21 +121,41 @@ import {
...
 
 
-// Create some content for the panel.
 let w1 = new Widget();
 let w2 = new Widget();
 let w3 = new Widget();
+```
 
+The size of the widgets inside a box panel can be set by means of the widget
+stretch factors, using relative sizes for each widget inside the box
```
Comment (sccolbert) [inline README.md]: s/relative sizes/relative weights

File: README.md [hunk 3]
```diff
@@ -117,21 +121,41 @@ import {
 } from 'phosphor-widget';
 
 
-// Create some content for the panel.
 let w1 = new Widget();
 let w2 = new Widget();
 let w3 = new Widget();
+```
 
+The size of the widgets inside a box panel can be set by means of the widget
```
Comment (sccolbert) [inline README.md]: s/The size of/The relative sizes of

File: README.md [hunk 4]
```diff
@@ -117,21 +121,41 @@ import {
...
 // Set the widget stretch factors (optional).
-BoxPanel.setStretch(w1, 0);
+BoxPanel.setStretch(w1, 1);
 BoxPanel.setStretch(w2, 2);
-BoxPanel.setStretch(w3, 1);
+BoxPanel.setStretch(w3, 3);
+```
+
+Another option is to set the size basis using `.setSizeBasis()` which takes as
+arguments the widget and the corresponding size in pixels.
```
Comment (sccolbert) [inline README.md]: Should add some more info like: "The size basis is used as the initial size of the widget before growing or shrinking to accommodate the available space."

File: README.md [hunk 5]
```diff
@@ -117,21 +121,41 @@ import {
...
 
+```typescript
 // Set the widget size basis (optional).
 BoxPanel.setSizeBasis(w1, 200);
 BoxPanel.setSizeBasis(w2, 650);
 BoxPanel.setSizeBasis(w3, 350);
+```
 
+A box panel arranges its children in a single row or column, but some
+parameters can be changed to customize the layout. `.direction` changes the
```
Comment (sccolbert) [inline README.md]: "changes the layout order of the widgets".

File: README.md [hunk 6]
```diff
@@ -117,21 +121,41 @@ import {
...
+direction, it is possible to select `BottomToTop`, `LeftToRight`, `RightToLeft`
```
Comment (sccolbert) [inline README.md]: s/direction, it/direction. It

File: README.md [hunk 7]
```diff
@@ -140,3 +164,41 @@ panel.addChild(w1);
...
+panel.attach(document.body);
+
+// sometime later...
+panel.detach();
+
+// or almost equivalently
+panel.dispose();
+```
+
+
```
Comment (sccolbert) [inline README.md]: These are all part of the base `Widget` class and should be documented there, not here.

File: README.md (hunk 1 above)
Comment (gochoam) [inline README.md]: Is it fine if I change that to "basis length of its children"?.

Comment (sccolbert) [inline README.md]: That doesn't make it any more clear to me. Can you describe what you are trying to say in a verbose fashion? Then we can pare it down.

Comment (gochoam) [inline README.md]: I'm trying to provide a quick description of what `setStretch` and `setSizeBasis` do without going into further details. Just as an introductory sentence.

File: README.md [hunk 8]
```diff
@@ -4,9 +4,9 @@ phosphor-boxpanel
 [![Build Status](https://travis-ci.org/phosphorjs/phosphor-boxpanel.svg)](https://travis-ci.org/phosphorjs/phosphor-boxpanel?branch=master)
 [![Coverage Status](https://coveralls.io/repos/phosphorjs/phosphor-boxpanel/badge.svg?branch=master&service=github)](https://coveralls.io/github/phosphorjs/phosphor-boxpanel?branch=master)
 
-A Phosphor layout panel which arranges its children into a single row or column.
-
-[API Docs](http://phosphorjs.github.io/phosphor-boxpanel/api/)
+This module provides a phosphor layout panel which arranges its children into a
+single row or column, making it possible to change the layout direction,
+spacing and the basis length of its children.
```
Comment (sccolbert) [inline README.md]: How about something like this?

"This module provides a phosphor layout panel which arranges its children into a single row or column. The user has control over layout direction, inter-element spacing, and preferred element size."

Comment (gochoam) [inline README.md]: Great!. I'll include that.

**Comment by gochoam** [comment]
@sccolbert I updated this to include the suggested edits. There is a problem tough, it failed the checks.
